spring-cloud-stream/src/main/java/org/springframework/cloud/stream/converter/JavaSerializationMessageConverter.java

```diff
@@ -16,44 +16,51 @@
 
 package org.springframework.cloud.stream.converter;
 
+import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
-import java.io.Serializable;
+import java.util.Arrays;
 
 import org.springframework.messaging.Message;
-
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.converter.AbstractMessageConverter;
 
 /**
- * A {@link org.springframework.messaging.converter.MessageConverter}
- * to convert from a POJO to byte[] with Java.io serialization if any.
- *
- * @author David Turanski
+ * @author Marius Bogoevici
  */
-public class JavaToSerializedMessageConverter extends AbstractFromMessageConverter {
+public class JavaSerializationMessageConverter extends AbstractMessageConverter {
 
-	public JavaToSerializedMessageConverter() {
-		super(MessageConverterUtils.X_JAVA_OBJECT, MessageConverterUtils.X_JAVA_SERIALIZED_OBJECT);
+	public JavaSerializationMessageConverter() {
+		super(Arrays.asList(MessageConverterUtils.X_JAVA_SERIALIZED_OBJECT, MessageConverterUtils.X_JAVA_OBJECT));
 	}
 
 	@Override
-	protected Class<?>[] supportedTargetTypes() {
-		return new Class<?>[] { byte[].class };
+	protected boolean supports(Class<?> clazz) {
+		return true;
```

Comment:
I subsequently corrected that. https://github.com/spring-cloud/spring-cloud-stream/pull/598/commits/94678ab17881ec711688eed8ddb3d97210393fd4#diff-3ee67112068a7c2632f85b9d9e290d5aR42